packages/ckeditor5-widget/src/widget.ts

```diff
@@ -207,22 +207,25 @@ export default class Widget extends Plugin {
 			// But at least triple click inside nested editable causes broken selection in Safari.
 			// For such event, we select the entire nested editable element.
 			// See: https://github.com/ckeditor/ckeditor5/issues/1463.
-			if ( ( env.isSafari || env.isGecko ) && domEventData.domEvent.detail >= 3 ) {
-				const mapper = editor.editing.mapper;
-				const viewElement = element.is( 'attributeElement' ) ?
-					element.findAncestor( element => !element.is( 'attributeElement' ) )! : element;
-				const modelElement = mapper.toModelElement( viewElement )!;
-
-				domEventData.preventDefault();
 
-				this.editor.model.change( writer => {
-					writer.setSelection( modelElement, 'in' );
-				} );
+			if ( ( env.isSafari || env.isGecko ) && domEventData.domEvent.detail >= 3 ) {
+				this._setSelectionInAncestorElement( element, domEventData );
 			}
 
 			return;
 		}
 
+		const parentElement = element.parent;
+
+		// If triple click should select entire paragraph.
+		if ( parentElement &&
+			parentElement.is( 'element' ) &&
+			!isWidget( parentElement ) &&
+			domEventData.domEvent.detail >= 3
+		) {
+			this._setSelectionInAncestorElement( element, domEventData );
+		}
```

Comment:
Why are checking the parent element? I'm not sure what this condition is checking.